core/controllers/topics_and_skills_dashboard.py

```
@@ -166,6 +166,107 @@ def get(self):
+        """Handles GET requests."""
+
+        skill_summaries = skill_services.get_all_skill_summaries()
+        skill_summary_dicts = [
+            summary.to_dict() for summary in skill_summaries]
+
+        skill_ids_assigned_to_some_topic = (
+            topic_fetchers.get_all_skill_ids_assigned_to_some_topic())
+        merged_skill_ids = (
+            skill_services.get_merged_skill_ids())
+
+        untriaged_skill_summary_dicts = []
+        categorized_skills_dict = {}
+        topics = topic_fetchers.get_all_topics()
+
+        uncategorized_skills_ids_and_topic_names = []
+        uncategorized_skill_ids = []
+        categorized_skill_ids_with_topic_and_subtopic_name = []
+        categorized_skill_ids = []
+
+        for topic in topics:
+            subtopics = topic.subtopics
+            categorized_skills_dict[topic.name] = {}
+            categorized_skills_dict[topic.name]['uncategorized'] = []
+
+            for skill_id in topic.uncategorized_skill_ids:
+                uncategorized_skill_ids.append(skill_id)
+                uncategorized_skills_ids_and_topic_names.append({
+                    'skill_id': skill_id,
+                    'topic_name': topic.name
+                })
+            for subtopic in subtopics:
+                categorized_skills_dict[topic.name][subtopic.title] = []
+                for skill_id in subtopic.skill_ids:
+                    categorized_skill_ids.append(skill_id)
+                    categorized_skill_ids_with_topic_and_subtopic_name.append({
+                        'skill_id': skill_id,
+                        'topic_name': topic.name,
+                        'subtopic_title': subtopic.title
+                    })
+
+        uncategorized_skills_descriptions = (
+            skill_services.get_descriptions_of_skills(
+                uncategorized_skill_ids)[0])
+        for skill_data in uncategorized_skills_ids_and_topic_names:
+            topic_name = skill_data.get('topic_name')
```

IMO this should be skill_data['topic_name']. "get()" introduces the possibility of None as a result, and in general, it's better to try and avoid branching like this. Similarly elsewhere.

That way, if something goes wrong, you at least get a traceback to tell you what's happening, rather than a silent failure.

---

done